certbot/main.py

```diff
@@ -859,6 +861,45 @@ def plugins_cmd(config, plugins):
     logger.debug("Prepared plugins: %s", available)
     notify(str(available))
 
+def enhance(config, plugins):
+    """Add security enhancements to existing configuration
+
+    :param config: Configuration object
+    :type config: interfaces.IConfig
+
+    :param plugins: List of plugins
+    :type plugins: `list` of `str`
+
+    :returns: `None`
+    :rtype: None
+
+    """
+    supported_enhancements = ["hsts", "redirect", "uir", "staple"]
+    # Check that at least one enhancement was requested on command line
+    if not any([getattr(config, enh) for enh in supported_enhancements]):
+        msg = ("Please specify one or more enhancement types to configure. To list "
+               "the available enhancement types, run:\n\n%s --help enhance\n")
+        logger.warning(msg, sys.argv[0])
+        raise errors.MisconfigurationError("No enhancements requested, exiting.")
+
+    try:
+        installer, _ = plug_sel.choose_configurator_plugins(config, plugins, "enhance")
+    except errors.PluginSelectionError as e:
+        return str(e)
+
+    if not config.certname:
+        config.certname = cert_manager.get_certnames(config, "enhance",
+                                                     allow_multiple=False)[0]
+    cert_domains = cert_manager.domains_for_certname(config, config.certname)
+    domain_question = ("Which domain names would you like to enable the selected "
+                       "enhancements for?")
+    domains = display_ops.choose_values(cert_domains, domain_question)
+    if not domains:
+        # To be consistent with the error messages (similar to get_certnames)
+        raise errors.Error("User ended interaction.")
+    le_client = _init_le_client(config, authenticator=None, installer=installer)
+    le_client.enhance_config(domains, config.chain_path, ask_redirect=False)
```

Comment:
`config.chain_path` isn't set unless it's included on the command line. I think we need to use `certbot.cert_manager.lineage_for_certname` to get the `RenewableCert` matching the cert name and use its `chain_path` property.

It may be worth adding logic to handle the case where `--chain-path` is also set.

Reply:
The `chain_path` from lineage is used if the user didn't provide `--chain-path` on CLI. The command line value overrides the one from lineage.